src/allmydata/test/test_node.py

```diff
@@ -207,123 +271,164 @@ def call_setLogDir(logdir):
             self.failUnless(isinstance(logdir, str), logdir)
         self.patch(foolscap.logging.log, 'setLogDir', call_setLogDir)
 
+        create_node_dir(basedir, "nothing to see here")
         TestNode(basedir)
         self.failUnless(ns.called)
 
-class EmptyNode(Node):
-    def __init__(self):
-        config = config_from_string("", "no portfile")
-        Node.__init__(self, config, 'no basedir')
-
-EXPECTED = {
-    # top-level key is tub.port category
-    "missing": {
-        # 2nd-level key is tub.location category
-        "missing": "alloc/auto",
-        "empty": "ERR2",
-        "disabled": "ERR4",
-        "hintstring": "alloc/file",
-        },
-    "empty": {
-        "missing": "ERR1",
-        "empty": "ERR1",
-        "disabled": "ERR1",
-        "hintstring": "ERR1",
-        },
-    "disabled": {
-        "missing": "ERR3",
-        "empty": "ERR2",
-        "disabled": "no-listen",
-        "hintstring": "ERR3",
-        },
-    "endpoint": {
-        "missing": "auto",
-        "empty": "ERR2",
-        "disabled": "ERR4",
-        "hintstring": "manual",
-        },
-    }
-
-class PortLocation(unittest.TestCase):
-    def test_all(self):
-        for tp in EXPECTED.keys():
-            for tl in EXPECTED[tp].keys():
-                exp = EXPECTED[tp][tl]
-                self._try(tp, tl, exp)
-
-    def _try(self, tp, tl, exp):
-        log.msg("PortLocation._try:", tp, tl, exp)
-        cfg_tubport = {"missing": None,
-                       "empty": "",
-                       "disabled": "disabled",
-                       "endpoint": "tcp:777",
-                       }[tp]
-        cfg_location = {"missing": None,
-                        "empty": "",
-                        "disabled": "disabled",
-                        "hintstring": "tcp:HOST:888,AUTO",
-                        }[tl]
-
-        n = EmptyNode()
-        basedir = os.path.join("test_node/portlocation/%s/%s" % (tp, tl))
-        fileutil.make_dirs(basedir)
-        config = n.config = read_config(
-            basedir,
-            "node.port",
-            _valid_config_sections=_valid_config_sections,
+
+class TestMissingPorts(unittest.TestCase):
+    """
+    Test certain error-cases for ports setup
+    """
+
+    def setUp(self):
+        self.basedir = self.mktemp()
+        create_node_dir(self.basedir, "testing")
+
+    def test_0(self):
```

Comment:
A better name and a docstring would be great here.  Same comment for the following tests.